base/iterators.jl

```diff
@@ -649,6 +649,51 @@ end
 iterate(it::Drop, state) = iterate(it.xs, state)
 isdone(it::Drop, state) = isdone(it.xs, state)
 
+
+# takewhile
+
+struct TakeWhile{I,P<:Function}
+    xs::I
+    pred::P
+end
+
+takewhile(pred,xs) = TakeWhile(xs,pred)
```

Comment:
The function should probably come first in the type constructor as well; this reversal is kind of weird.